lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp

```diff
@@ -0,0 +1,71 @@
+#include "mlir/Bytecode/BytecodeOpInterface.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/OperationSupport.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
+#include "triton/Dialect/Triton/IR/Types.h"
+#include "llvm/ADT/STLExtras.h"
+
+#define GET_OP_CLASSES
+#include "triton-shared/Dialect/TritonStructured/IR/TritonStructuredOps.h.inc"
+
+using namespace mlir;
+using namespace mlir::tts;
+
+namespace mlir {
+namespace tts {
+
+void MakeTensorPtrOp::build(OpBuilder &b, OperationState &state, Value base,
+                            ArrayRef<int64_t> sizes,
+                            ArrayRef<OpFoldResult> strides,
+                            ArrayRef<OpFoldResult> offsets,
+                            ArrayRef<OpFoldResult> parentSizes) {
+  SmallVector<int64_t> staticStrides, staticOffsets, staticParentSizes;
+  SmallVector<Value> dynamicStrides, dynamicOffsets, dynamicParentSizes;
+
+  dispatchIndexOpFoldResults(offsets, dynamicOffsets, staticOffsets);
+  dispatchIndexOpFoldResults(strides, dynamicStrides, staticStrides);
+  dispatchIndexOpFoldResults(parentSizes, dynamicParentSizes,
+                             staticParentSizes);
+
+  auto basePtr = base.getType().cast<triton::PointerType>();
```

Comment:
nit: mlir is looking to at one point deprecate this style of casting, prefer to use `cast<triton::PointerType>(base.getType())` instead 😄